### PR TITLE
chore: upgrade dd-trace to maintenance version

### PR DESCRIPTION
**Pull request by ovasdi**
The type of this PR is: **Chore**

This PR solves [PLATFORM-5144](https://artsyproduct.atlassian.net/browse/PLATFORM-5144) and is a follow-up to https://github.com/artsy/force/pull/12328

### Description

This upgrades dd-trace library to v2.31.0, the latest "Maintenance" version ([version table](https://github.com/DataDog/dd-trace-js/tree/master#version-release-lines-and-maintenance)).

Looking at [v1 -> v2 migration doc](https://github.com/DataDog/dd-trace-js/blob/master/MIGRATING.md#10-to-20) it looks to me like none of the migration changes apply to Force:

- Force does not use the removed Configuration options: `enabled`, `debug`
- Force does not appear to set `DD_TRACE_ENABLED` env var however when instrumenting the library, tracer is [enabled by default](https://docs.datadoghq.com/tracing/trace_collection/library_config/nodejs/#instrumentation).
- configuration option `scope` and the removed API `tracer.scopeManager()` are not used in Force
- Force does not appear to set [nested object as a tag value](https://github.com/DataDog/dd-trace-js/blob/master/MIGRATING.md#nested-objects-as-tags). The only usage of `setTag` is [here](https://github.com/search?q=repo%3Aartsy%2Fforce%20span.setTag&type=code) and the value is a string
- Not clear on the [outgoing-request-filtering](https://github.com/DataDog/dd-trace-js/blob/master/MIGRATING.md#outgoing-request-filtering) but it doesn't appear to affect Force

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PLATFORM-5144]: https://artsyproduct.atlassian.net/browse/PLATFORM-5144?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ovasdi** [comment]
As I am attempting to upgrade this lib to a maintained version, I ran into a type error that I'm not sure how to resolve. The [type error](https://app.circleci.com/pipelines/github/artsy/force/47991/workflows/46e348a6-a186-461b-bc72-d0484b5a7191/jobs/360441?invite=true#step-101-10) is coming from dd-tracer hook config where we are doing some stuff with the [request](https://github.com/artsy/force/blob/main/src/Server/datadog.ts#L25). I am looking at the types for [v0.36.2](https://github.com/DataDog/dd-trace-js/blob/v0.36.6/index.d.ts) and [v1.7.1](https://github.com/DataDog/dd-trace-js/blob/v1.7.1/index.d.ts) and am not able to discern whats different.
Perhaps its safe to ignore this with `// @ts-ignore` ?

**Comment by ovasdi** [comment]
> As I am attempting to upgrade this lib to a maintained version, I ran into a type error that I'm not sure how to resolve. The [type error](https://app.circleci.com/pipelines/github/artsy/force/47991/workflows/46e348a6-a186-461b-bc72-d0484b5a7191/jobs/360441?invite=true#step-101-10) is coming from dd-tracer hook config where we are doing some stuff with the [request](https://github.com/artsy/force/blob/main/src/Server/datadog.ts#L25). I am looking at the types for [v0.36.2](https://github.com/DataDog/dd-trace-js/blob/v0.36.2/index.d.ts) and [v1.7.1](https://github.com/DataDog/dd-trace-js/blob/v1.7.1/index.d.ts) and am not able to discern whats different. Perhaps its safe to ignore this with `// @ts-ignore` ?

This [line](https://github.com/DataDog/dd-trace-js/pull/2982/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R874) introduced the type change in release v2.31.0

**Comment by mc-jones** [comment]
LGTM! Still WIP?

**Comment by ovasdi** [comment]
> LGTM! Still WIP?

Yes ready now.

File: src/Server/datadog.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
-import ddTracer from "dd-trace"
+import ddTracer, { Span } from "dd-trace"
```
Comment (ovasdi) [inline src/Server/datadog.ts:1]: This fixed a type change that was introduced in v2.31.0. See [this](https://artsy.slack.com/archives/CA8SANW3W/p1682707134740739?thread_ts=1682540612.539359&cid=CA8SANW3W) slack comment for more info.

**Review by mc-jones — APPROVED** [review]
🙌

**Pull request merged**